plugins/modules/gcp_compute_instance.py

```diff
@@ -1715,6 +1729,7 @@ def _request_for_item(self, item):
                 u'network': replace_resource_dict(item.get(u'network', {}), 'selfLink'),
                 u'networkIP': item.get('network_ip'),
                 u'subnetwork': replace_resource_dict(item.get(u'subnetwork', {}), 'selfLink'),
+                u'nicType': item.get('nic_type'),
```

Comment:
So if the value of nicType is None it will deploy with the default?

Reply:
If so I'd like to see a test case below that tests with nicType being omitted.

Reply:
ack, reworked the test cases, there's a caveat: if you don't specify the nic_type parameter then the API won't expose it back